uaa/src/test/java/org/cloudfoundry/identity/uaa/mock/zones/ZoneControllerMockMvcTests.java

```diff
@@ -0,0 +1,249 @@
+        BaseClientDetails uaaAdminClient = new BaseClientDetails("uaa-admin-" + RandomString.make(5).toLowerCase(),
+                                                                 null,
+                                                                 "uaa.admin",
+                                                                 "password,client_credentials",
+                                                                 "uaa.admin");
+        uaaAdminClient.setClientSecret("secret");
+        clientRegistrationService.addClientDetails(uaaAdminClient);
+        identityClientZonesReadToken = testClient.getClientCredentialsOAuthAccessToken(
+            "identity",
+            "identitysecret",
+            "zones.read");
+        uaaAdminClientToken = testClient.getClientCredentialsOAuthAccessToken(
+            uaaAdminClient.getClientId(),
+            "secret",
+            "uaa.admin");
+    }
+
+    @ParameterizedTest
+    @ArgumentsSource(IdentityZonesBaseUrlsArgumentsSource.class)
+    void testGetZone_unAuthorized(String url) throws Exception {
+        mockMvc.perform(get(url))
+               .andExpect(status().isUnauthorized());
+    }
+
+    @ParameterizedTest
+    @ArgumentsSource(NameRequiredArgumentsSource.class)
+    void testGetZone_nameRequiredError(String url) throws Exception {
+        MvcResult result = mockMvc.perform(
+                                      get(url)
+                                          .header("Authorization", "Bearer " + identityClientZonesReadToken))
+                                  .andExpect(status().isBadRequest()).andReturn();
+        assertEquals(
+            "{\"message\", \"Required request parameter 'name' for method parameter type String is not present\" }",
+            result.getResponse().getContentAsString());
+    }
+
+    @ParameterizedTest
+    @ArgumentsSource(NameNotEmptyArgumentsSource.class)
+    void testGetZone_nameEmptyError(String url) throws Exception {
+        MvcResult result = mockMvc.perform(
+                                      get(url)
+                                          .header("Authorization", "Bearer " + identityClientZonesReadToken))
+                                  .andExpect(status().isBadRequest()).andReturn();
+        assertEquals(
+            "{\"message\", \"getZone.name: must not be empty\" }",
+            result.getResponse().getContentAsString());
+    }
+
+    @Test
+    void testGetZone() throws Exception {
+        //TODO: remove this createIdentityZone method once the create orchestrator zone API is implemented and use it to create zone.
+        IdentityZone identityZone = createIdentityZone();
+        MvcResult result = mockMvc.perform(
+                   get("/zones").param("name", "test-name")
+                                .header("Authorization", "Bearer " + identityClientZonesReadToken))
+               .andExpect(status().isAccepted()).andReturn();
+        if (HttpStatus.OK.is2xxSuccessful()) {
```

Comment:
Why if condition? for successful GET, it should always validate response.

Reply:
Removed.